lib/ansible/modules/cloud/azure/azure_rm_autoscale.py

```diff
@@ -0,0 +1,620 @@
+                - This element is not used if the FixedDate element is used.
+            recurrence_mins:
+                description:
+                - the mins of repeating times at which this profile begins.
+                - This element is not used if the FixedDate element is used.
+            fixed_date_timezone:
+                description:
+                - the specific date-time timezone for the profile.
+                - This element is not used if the Recurrence element is used.
+            fixed_date_start:
+                description:
+                - the specific date-time start for the profile.
+                - This element is not used if the Recurrence element is used.
+            fixed_date_end:
+                description:
+                - the specific date-time end for the profile.
+                - This element is not used if the Recurrence element is used.
+            rules:
+                description:
+                - The collection of rules that provide the triggers and parameters for the scaling action.
+                - A maximum of 10 rules can be specified.
+                suboptions:
+                    time_aggregation:
+                        default: Average
+                        description: How the data that is collected should be combined over time.
+                        choices:
+                        - Average
+                        - Minimum
+                        - Maximum
+                        - Total
+                        - Count
+                    time_window:
+                        required: true
+                        description:
+                        - The range of time(minutes) in which instance data is collected.
+                        - This value must be greater than the delay in metric collection, which can vary from resource-to-resource.
+                        - Must be between 5 ~ 720.
+                    direction:
+                        description: Whether the scaling action increases or decreases the number of instances.
+                        choices:
+                        - Increase
+                        - Decrease
+                    metric_name:
+                        required: true
+                        description: The name of the metric that defines what the rule monitors.
+                    metric_resource_uri:
+                        description: The resource identifier of the resource the rule monitors.
+                    value:
+                        description:
+                        - The number of instances that are involved in the scaling action.
+                        - This value must be 1 or greater.
+                    operator:
+                        default: GreaterThan
+                        description: The operator that is used to compare the metric data and the threshold.
+                        choices:
+                        - Equals
+                        - NotEquals
```

Comment:
i think we should have these choices in python case perhaps "not_equals" etc....?
that applies to all options.

Reply:
Actually, since the default is str(None), the Null can be convert as str(None)